lib/logger.py

```diff
@@ -0,0 +1,132 @@
+#
+#  Multi-stream logging with timestamps
+#
+# This is a simple logger for storing several independent data streams. It was
+# developed during Naio competition (see https://github.com/robotika/naio) as
+# an alternative to multiple log files. The concept is similar to ADTF (Automotive
+# Data and Time-Triggered Framework) DAT files.
+#
+#   The file is using Python delta time structures for timestamps. Each data
+# block has header of form:
+#   <timestamp>, <streamID>, <size of data>, <data>
+# where "timestamp" is 32bit unsigned integer in microseconds, "streamID" is
+# 16bit unsigned integer defining stream (0 is reserved for info, input and
+# output have different ID), and "size" is 16bit unsigned integer defining
+# the size of followed data.
+#
+#   The choice of microseconds was compromise between necessary number of bytes
+# for timestamp and available resolution. 32bit corresponds to 4294 seconds, i.e.
+# more than an hour of recording. The file contains absolute time and date in
+# the overall file header allowing splitting the file and getting absolute time
+# from deltas if necessary.
+#
+#   The stream ID is currently just integer without detailed description. There
+# is planned extra info stored in "info channel/stream" ID = 0.
+#
+#   Finally limiting the block size to 65kB can be overcome by pre-reserved value
+# 0 and 0xFFFF. The big block could be then split into several smaller once. This
+# part is not defined yet.
+#
+
+import datetime
+import struct
+from threading import Lock
+
+
+INFO_STREM_ID = 0
+
+
+class LogWriter:
+    def __init__(self, prefix='naio', note=''):
+        self.lock = Lock()
+        self.start_time = datetime.datetime.now()
```

Comment:
`datetime.now` return current local time. Daylight savings and different timezones might result in unwanted differences during replay. It would be better to store UTC time inside the log, thus have unambiguous definition of `now`.

Reply:
OK, I am changing it to `utcnow()` now ... it would need some test ...